File: src/test/java/me/saro/jwt/java/io/JwtObjectBuilderTest.java
package me.saro.jwt.java.io;

import io.jsonwebtoken.SignatureAlgorithm;
import me.saro.jwt.old.impl.DefaultKeyChain;
import me.saro.jwt.old.io.JwtBuilder;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

@DisplayName("[Java] JwtBuilder")
public class JwtObjectBuilderTest {
    @Test
    @DisplayName("arguments check")
    public void t2() {
        Assertions.assertThrows(IllegalArgumentException.class, () -> new JwtBuilder(SignatureAlgorithm.RS384, DefaultKeyChain.create(SignatureAlgorithm.RS256)));

        JwtBuilder builder = new JwtBuilder(SignatureAlgorithm.RS256, DefaultKeyChain.create(SignatureAlgorithm.RS256));

        Assertions.assertThrows(IllegalArgumentException.class, () -> builder.header("kid", ""));
        Assertions.assertThrows(IllegalArgumentException.class, () -> builder.header("alg", ""));
        Assertions.assertThrows(IllegalArgumentException.class, () -> builder.header("", ""));

        Assertions.assertThrows(IllegalArgumentException.class, () -> builder.claim("exp", ""));
        Assertions.assertThrows(IllegalArgumentException.class, () -> builder.claim("iat", ""));
        Assertions.assertThrows(IllegalArgumentException.class, () -> builder.claim("", ""));
    }

    @Test
    @DisplayName("build")
    public void t3() {
        Assertions.assertDoesNotThrow(() -> {
            JwtBuilder builder = new JwtBuilder(SignatureAlgorithm.RS256, DefaultKeyChain.create(SignatureAlgorithm.RS256));
            builder.claim("name", "안녕");
            builder.build();
        });
    }
}

File: src/test/kotlin/me/saro/jwt/kotlin/io/JwtObjectBuilderTest.kt
package me.saro.jwt.kotlin.io

import io.jsonwebtoken.SignatureAlgorithm
import me.saro.jwt.old.impl.DefaultKeyChain
import me.saro.jwt.old.io.JwtBuilder
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("[Kotlin] JwtBuilder")
class JwtObjectBuilderTest {
    @Test
    fun `arguments check`() {
        Assertions.assertThrows(IllegalArgumentException::class.java) { JwtBuilder(SignatureAlgorithm.RS384, DefaultKeyChain.create(SignatureAlgorithm.RS256)) }

        val builder = JwtBuilder(SignatureAlgorithm.RS256, DefaultKeyChain.create(SignatureAlgorithm.RS256))

        Assertions.assertThrows(IllegalArgumentException::class.java) { builder.header("kid", "") }
        Assertions.assertThrows(IllegalArgumentException::class.java) { builder.header("alg", "") }
        Assertions.assertThrows(IllegalArgumentException::class.java) { builder.header("", "") }
        Assertions.assertThrows(IllegalArgumentException::class.java) { builder.claim("exp", "") }
        Assertions.assertThrows(IllegalArgumentException::class.java) { builder.claim("iat", "") }
        Assertions.assertThrows(IllegalArgumentException::class.java) { builder.claim("", "") }
    }

    @Test
    fun build() {
        Assertions.assertDoesNotThrow {
            val builder = JwtBuilder(SignatureAlgorithm.RS256, DefaultKeyChain.create(SignatureAlgorithm.RS256))
            builder.claim("name", "안녕")
            builder.build()
        }
    }
}